### PR TITLE
Add e2e tests for tpu-provisioner

### PR DESCRIPTION
**Pull request by nstogner**
Creates a cluster, installs tpu-provisioner, and then runs test JobSets against the cluster.

### CODE REVIEW

File: tpu-provisioner/test/e2e/test/jobset_test.go [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+		}
+
+		tpu_v5e_2x4 = tpuConfig{
+			accelerator:  "tpu-v5-lite-podslice",
+			topoX:        2,
+			topoY:        4,
+			chipsPerNode: 4,
+			sliceCount:   2,
+		}
+	*/
```
Comment (danielvegamyhre) [inline tpu-provisioner/test/e2e/test/jobset_test.go:232]: Are these commented out intentionally or was it meant to be temporarily commented out for your testing?

File: tpu-provisioner/test/e2e/test/jobset_test.go [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+	z := int32(1)
+	if t.topoZ > 0 {
+		z = t.topoZ
+	}
+	return t.topoX * t.topoY * z / t.chipsPerNode
+}
+
+func newJobset(name string, c tpuConfig, uniqueNodeSelector bool) *jobset.JobSet {
+	nodeSelectors := map[string]string{
+		"cloud.google.com/gke-tpu-accelerator": c.accelerator,
```
Comment (danielvegamyhre) [inline tpu-provisioner/test/e2e/test/jobset_test.go:275]: Can we define the various labels in this file as constants somewhere

File: tpu-provisioner/test/e2e/test/util/cleanup.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, os.Interrupt)
+	go cleanupOnInterrupt(c)
+}
+
+var cf struct {
+	o sync.Once
+	m sync.RWMutex
+	f []func()
+}
```
Comment (danielvegamyhre) [inline tpu-provisioner/test/e2e/test/util/cleanup.go:38]: The variable names for this struct and its fields are unclear to me, can we make them more descriptive?

I'm guessing:

"cf" = ?
 "o" = "once"
 "m" = "mutex"
 "f" = "functions"

**Comment by spencer-p** [comment]
Closing old issues with merge conflicts. Please rebase and re-open if still relevant.

**Pull request closed without merging**